src/addons/grids/ReactDataGrid.js

```diff
@@ -93,12 +93,21 @@ var ReactDataGrid = React.createClass({
     var initialState = {selectedRows : [], copied : null, expandedRows : [], canFilter : false, columnFilters : {}, sortDirection: null, sortColumn: null, dragged : null}
     if(this.props.enableCellSelect){
       initialState.selected = {rowIdx: 0, idx: 0};
+      initialState.selectedRows = this.getInitialSelectedRows();
     }else{
```

Comment:
this should be outside check for enableCellSelect
That is something different, and not always set
Should be  var `initialState = {selectedRows : this.getInitialSelectedRows(),...`